### ISSUE TITLE
Cinnamon and Mate: thirty-second delay when running some flatpaks; an indication of what *the fix* is.

### ISSUE THREAD

**Issue by LinuxOnTheDesktop** [linuxmint/cinnamon#9373]
```
 * Cinnamon version (cinnamon --version): 4.4.8
   - Please specify if you are using the daily builds PPA: no
     (https://launchpad.net/~linuxmint-daily-build-team/+archive/ubuntu/daily-builds).
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...): Mint 19.3
 * Graphics hardware *and* driver used: various (problem affects >1 computer)
 * 32 or 64 bit: 64
 * Attach /home/<username>/.xsession-errors, or /var/log/syslog: only one line of the syslog seems relevant. It's this: 'dbus-daemon	[session uid=0 pid=30081] Failed to activate service 'org.a11y.Bus': timed out (service_start_timeout=120000ms)'
 ```

**Issue**

On two computers, one of them fast (quad-core, nvme, 16GB RAM), the flatpak of `fwupdmgr` takes some thirty seconds to do anything including show its help text. I have reason to think that the same problem besets other flatpaks and indeed affects Mate as well as Cinnamon. On those matters please see below.

**Steps to reproduce**

```
sudo flatpak install flathub org.freedesktop.fwupd
sudo flatpak run org.freedesktop.fwupd
```

**Expected behaviour**

It should take at most a second or two to start the application.

**Other information**

- I am unsure what other flatpaked applications are affected. Yet, there are reports of the problem occurring, with multiple flatpak applications, on [Arch with Plasma (report on the flatpak bug tracker)](https://github.com/flatpak/flatpak/issues/2843) and on [Mint with Mate (report on Reddit)](https://www.reddit.com/r/linuxmint/comments/cqo6fh/bug_flatpak_on_linux_mint_mate_192_starts_really/).

- The problem does not occur after `su` has been issued in the terminal.

- The [log from fwupd is here](https://gist.github.com/LinuxOnTheDesktop/eadd9f19649890fd766cd01565191b8a).

- **_Plasma managed to fix the problem_** - see [here](https://bugs.kde.org/show_bug.cgi?id=406672).

- I filed a bug with flatpak but [it](https://github.com/flatpak/flatpak/issues/3653) has had no response as yet (although there is also [the aforementioned bug report, filed against flatpak, from someone using Plasma on Arch Linux](https://github.com/flatpak/flatpak/issues/2843); and that report links to the solution that, as I wrote above, Plasma has adopted).

- I reported the bug to fwup (see towards the end of [here](https://github.com/fwupd/fwupd/issues/2156)) but the people there do not think it is their problem.

**Comment by Nokia808** [comment]
@LinuxOnTheDesktop 
Yes, I agree with you ! I suffer from this since I was on Fedora 30. After upgraded to Fedora 32 there was improvement when I being on flatpak version 1.6.3

But few weeks after upgrade - about 2 weeks - when I start to suffer from re-occurrence of slow Cinnamon launching time, I started to suffer again from slow start of flatpak !

All flatpak affected, but most sever one is LibreOffice: 3 minutes to start !!

I think that this issue is related to the following issue https://github.com/linuxmint/cinnamon/issues/9356

**Comment by LinuxOnTheDesktop** [comment]
@Nokia808 

I found a flatpak that was not affected, namely, Krita.

[It seems](https://packages.ubuntu.com/search?suite=focal&keywords=flatpak) that Ubuntu 20 LTS ('Focal') has flatpak version 1.6.3-1. So perhaps with Mint 20 the problem will be solved. I do not myself fancy trying, and certainly not actually installing, the newly released beta of Mint 20. But perhaps someone could do that, and could test its flatpak with e.g. fwupd.

**Comment by LinuxOnTheDesktop** [comment]
Just now I tried the flatpak of fwup on a virtual machine (running on Windows) of Mint 19.3. It had the problem of the very slow start.

I will try Mint 20 Cinnamon beta in a VM and see whether that works.

**Comment by LinuxOnTheDesktop** [comment]
I just tried installing the flatpak version of fwup on a virtualised live version Mint 20 Cinnamon beta. It worked and fwup started quickly.

I installed the beta - but still in the VM - and again fwupd ran quickly from the flatpak. I just hope - given that a report above says that the problem can be intermittent - that it stays that way.

**Comment by LinuxOnTheDesktop** [comment]
I experience this problem no longer.

**Comment by Nokia808** [comment]
Hi. I do not think this issue is solved if you are talking about version 4.8.6

Now, I'm on Fedora 34 X64 bit Cinnamon edition. To be more clear & to not causing confusion, I have to inform you that I'm upgraded my RAM from 8GB to 16GB. At 8GB RAM copy/past of archive file from my Lenovo ThinkPad e550 (HDD 1TB) to Samsung 870 OVE SSD 2TB SATA3 set in Orico external SATA3 enclosure & connected to USB 3.0 was 55 second. Now, with 16GB RAM this copy/past process need only 14 second to be completed.

I tried to open my flatpaks, & I noticed that 1st launching process, after power on laptop, taking from 10 seconds to 13 seconds. But subsequent launching on same session taking only 2 seconds ! So, this issue is still existing & not fixed, at least with Cinnamon version 4.8.6

Kindly, review the above data.
